### PR TITLE
fix(multi-action-button, split-button): ensure screen reader commands can navigate menu popup

### LINKED ISSUE #7054: Multi Action Button still not always working for screenreader despite being tabbable - serious
### Description

When I click on dynamic shortcut button in home.ui (eg. Go to Accounting, Go to Payroll...) with a screenreader, it should focus on the options instead of ignoring it and going through the rest of the DOM.

Similar to this issue [here](https://github.com/Sage/carbon/issues/7005), I have tried on updated carbon but this still does not work despite being tabbable

https://github.com/user-attachments/assets/7905501d-935c-48d2-addd-995124a14018

### Reproduction

https://carbon.sage.com/?path=/story/multi-action-button--button-types

### Steps to reproduce

Open screenreader, select multi-action button, click to open and then go to next element

### JIRA ticket numbers (Sage only)

SBS-113601

### Suggested solution

_No response_

### Carbon version

144.4.0

### Design tokens version

4.34.0

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by nuria1110**
fix #7054

### Proposed behaviour

 
- First option in menu popup gains focus when menu is open. 
- Existing keyboard navigation Space, Enter and Arrow Up/Down should still work as expected. 
- Menu should close when no option has focus. 
- Menu popup is no longer in a portal.

### Current behaviour

- Screen reader users using VoiceOver commands cannot navigate into the menu popup. 
- Menu popup in in a portal. 

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Best tested in an example containing several focusable elements e.g.: "Sizes" story. 
- Navigate to toggle button and use VO commands Control+Option+Space to open menu. 
- Use Control+Option+Right to navigate menu options - see VO move onto next button on the page instead of the menu.

### CODE REVIEW

**Review by DipperTheDan — APPROVED** [review]

File: src/components/split-button/split-button.test.tsx [hunk 1]
```diff
@@ -679,49 +643,6 @@ test("should hide the additional buttons when a 'Escape' keydown event detected
   expect(screen.queryByRole("list")).not.toBeInTheDocument();
 });
 
-test("should render with the correct styles when 'align' prop is 'right'", async () => {
```
[the remark this answers is not in the document]
Comment (nuria1110) [reply src/components/split-button/split-button.test.tsx:682]: Removed these as they seemed to be pretty much the same as the tests [here](https://github.com/Sage/carbon/blob/c810d5ce38d8b062dfb061a1af5787e38c2b0ed2/src/components/split-button/split-button.test.tsx#L368)

[the remark this answers is not in the document]
Comment (nuria1110) [reply src/components/split-button/split-button.test.tsx:682]: Also looks like we have PW tests too

[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/split-button/split-button.test.tsx:682]: nice one thanks for checking

**Review by DipperTheDan — APPROVED** [review]

**Comment by harpalsingh** [comment]
I have tested the VO and works as expected now, so will pass a11y review too.

**Pull request merged**